gpustack/routes/model_files.py

```diff
@@ -130,6 +131,24 @@ async def create_model_file(session: SessionDep, model_file_in: ModelFileCreate)
             message="Model file with the same model source already exists on the worker."
         )
 
+    if model_file_in.local_dir is not None:
+        fields = {
+            "worker_id": model_file_in.worker_id,
+            "local_dir": model_file_in.local_dir,
+        }
+        worker_existing_files = await ModelFile.all_by_field(
+            session, field="worker_id", value=model_file_in.worker_id
+        )
+        if worker_existing_files:
+            for file in worker_existing_files:
+                if (
+                    file.local_dir
+                    and Path(file.local_dir).resolve()
+                    == Path(model_file_in.local_dir).resolve()
+                ):
+                    raise AlreadyExistsException(
+                        message=f"Local directory {model_file_in.local_dir} already in use on the worker."
```

Comment:
It seems to be a valid use case to download multiple GGUF models in the same directory